cli/azd/pkg/apphost/generate.go

```diff
@@ -274,6 +286,60 @@ func newInfraGenerator() *infraGenerator {
 	}
 }
 
+func evaluateForOutputs(value string) (map[string]genOutputParameter, error) {
+	outputs := make(map[string]genOutputParameter)
+
+	re := regexp.MustCompile(`\{[a-zA-Z0-9\-]+\.(outputs|secretOutputs)\.[a-zA-Z0-9\-]+\}`)
+	matches := re.FindAllString(value, -1)
+	for _, match := range matches {
+		noBrackets := strings.TrimRight(strings.TrimLeft(match, "{"), "}")
+		parts := strings.Split(noBrackets, ".")
+		name := fmt.Sprintf("%s_%s", strings.ToUpper(parts[0]), strings.ToUpper(parts[2]))
+		outputType := "outputs"
+		if strings.Contains(match, "secretOutputs") {
+			outputType = "secretOutputs"
+		}
+		outputs[name] = genOutputParameter{
+			Type:  "string",
+			Value: fmt.Sprintf("%s.%s.%s", parts[0], outputType, parts[2]),
+		}
+	}
+	return outputs, nil
+}
+
+// extractOutputs evaluates the known fields where a Resource can reference an output and persist it.
+func (b *infraGenerator) extractOutputs(resource *Resource) error {
+	// from connection string
+	if resource.ConnectionString != nil {
+		outputs, err := evaluateForOutputs(*resource.ConnectionString)
+		if err != nil {
+			return err
+		}
+		for key, output := range outputs {
+			if strings.Contains(output.Value, ".outputs.") {
```

Comment:
pushing the "secretness" into the actual output object itself would let us remove these if forks (and we already have a version of this if fork in `evaluateForOutputs` since the value it writes differs, so I think we wouldn't even need to add new control flow to do it.

Reply:
the complexity of validating for secret or not secret is moved to the go-template.